src/cryptonote/hash.rs

```diff
@@ -99,17 +104,25 @@ impl crate::consensus::encode::Encodable for Hash {
     }
 }
 
+/// Errors encountered when encoding or decoding data.
+#[derive(Error, Debug, PartialEq)]
+pub enum HashError {
+    /// Encoding error.
+    #[error("Encode error: {0}")]
+    EncodeError(#[from] EncodeError),
+}
+
 /// Capacity of an object to hash itself and return the result as a plain [`struct@Hash`] or as an
 /// interpreted scalar value into [`PrivateKey`].
 pub trait Hashable {
     /// Return its own hash.
-    fn hash(&self) -> Hash;
+    fn hash(&self) -> Result<Hash, HashError>;
```

Comment:
This does complicate the API on something that "shouldn't" fail? maybe im missing something again?

Reply:
Please see unit test `fn previous_fuzz_transaction_hash_failures()` in #182